2020-02-08 Jiaxuan Xue <nimbusxue1015@g.ucla.edu>

* timespec.h(timespec_fix.h):Change comparison method to avoid overflow       
diff -u lib/timespec.h lib/timespec_fix.h 

--- lib/timespec.h	2005-09-21 23:05:39.000000000 -0700
+++ lib/timespec_fix.h	2020-02-08 21:28:21.435730000 -0800
@@ -45,8 +45,12 @@
 static inline int
 timespec_cmp (struct timespec a, struct timespec b)
 {
-  int diff = a.tv_sec - b.tv_sec;
-  return diff ? diff : a.tv_nsec - b.tv_nsec;
+  if(a.tv_sec > b.tv_sec)
+    return 1;
+  if(a.tv_sec < b.tv_sec)
+    return -1;
+  if(a.tv_sec == b.tv_sec)
+    return a.tv_nsec - b.tv_nsec;
 }
 
 # if ! HAVE_DECL_NANOSLEEP
